CRM/Contribute/BAO/Contribution.php

```diff
@@ -5457,4 +5457,25 @@ public static function createProportionalFinancialEntries($entityParams, $lineIt
     }
   }
 
+  /**
+   * Calculate Tax for each item when Financial Type is changed.
+   *
+   * @param array $lineItem
+   *
+   * @param int $contributionId
+   *
+   */
+  public static function calculateTaxAfterChangeInFinancialTypeForLineItems($lineItem, $contributionId) {
```

Comment:
I think this method name and its description are confusing. Usually, the method name should describe what it does and not when it should be executed. It says it calculates the Tax after the Financial Type is changed, but looking at the code, it looks like it can do the calculation anytime, even if the Financial Type hasn't been changed.